soroban-env-host/src/auth.rs

```diff
@@ -1135,10 +1203,14 @@ impl InvocationTracker {
         None
     }
 
-    fn push_frame(&mut self) {
+    // metering: covered
+    fn push_frame(&mut self, budget: &Budget) -> Result<(), HostError> {
```

Comment:
I _think_ we already account for this as a part of the `GuardFrame`, so this is technically charged twice.

Reply:
You are right.
In this case I think `GuardFrame` is no longer necessary. It merely does a context frame push + auth frame push, each of which can (and should) be metered individually. I will clean this part up, and remove the `xdr::GuardFrame`.

Reply:
Yeah, that sounds good, thanks.

Reply:
Okay, I've cleaned up metering of `GuardFrame`. Will do the xdr removal in a follow up.